route/otlp_trace.go

```diff
@@ -65,18 +64,27 @@ func processTraceRequest(
 		return err
 	}
 
-	for _, ev := range batch {
-		event := &types.Event{
-			Context:    ctx,
-			APIHost:    apiHost,
-			APIKey:     apiKey,
-			Dataset:    datasetName,
-			SampleRate: uint(ev.SampleRate),
-			Timestamp:  ev.Timestamp,
-			Data:       ev.Attributes,
-		}
-		if err = router.processEvent(event, requestID); err != nil {
-			router.Logger.Error().Logf("Error processing event: " + err.Error())
+	// get environment name - will be empty for legacy keys
```

Comment:
it will be `$legacy$` for legacy keys, right? does that change anything downstream?